cmd/otelcontribcol/go.mod

```diff
@@ -4,7 +4,7 @@ module github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontrib
 
 go 1.21.0
 
-toolchain go1.21.12
+toolchain go1.22.5
```

Comment:
This is my first PR review after the (long) PTO, but perhaps this change shouldn't be here?